packages/flag-evaluation/src/index.ts

```diff
@@ -106,49 +226,40 @@ export function flattenJSON(data: object): Record<string, string> {
   return result;
 }
 
-export function unflattenJSON(data: Record<string, any>) {
+/**
+ * Converts a flattened JSON object with dot-separated keys into a nested JSON object.
+ *
+ * @param {Record<string, any>} data - The flattened JSON object where keys are dot-separated representing nested levels.
+ * @return {Record<string, any>} The unflattened JSON object with nested structure restored.
+ */
+export function unflattenJSON(data: Record<string, any>): Record<string, any> {
   const result: Record<string, any> = {};
 
   for (const i in data) {
     const keys = i.split(".");
     keys.reduce((acc, key, index) => {
       if (index === keys.length - 1) {
+        assert(typeof acc === "object", "overlapping keys");
```

Comment:
It will explodat anyway even without this assert because we would try to set a property on a string and that will trigger an error. Should I simply skip this assignment in this case to make it not throw?

Reply:
ah, i see. Yeah, let's simply skip it 👍